src/geowombat/core/sops.py

```diff
@@ -1116,66 +1127,103 @@ def coregister(target, reference, **kwargs):
+                        {
+                            'crs': ref_src.gw.crs_to_pyproj.to_wkt(version=wkt_version)
+                        }
+                    )
+                if 'nodata' in kwargs:
+                    ref_src = ref_src.assign_attrs({'nodatavals': (kwargs['nodata'][0],)}).fillna(kwargs['nodata'][0])
+                ref_src.gw.save(ref_path, overwrite=True)
+            with gw_.open(target.filename, chunks={'band': -1, 'y': 256, 'x': 256}) as tar_src:
+                if wkt_version is not None:
+                    tar_src = tar_src.assign_attrs(
+                        {
+                            'crs': tar_src.gw.crs_to_pyproj.to_wkt(version=wkt_version)
+                        }
+                    )
+                if 'nodata' in kwargs:
+                    tar_src = tar_src.assign_attrs({'nodatavals': (kwargs['nodata'][1],)}).fillna(kwargs['nodata'][1])
+                tar_src.gw.save(tar_path, overwrite=True)
+
+            cr = arosics.COREG(
+                GeoArray(GeoArray(str(ref_path)), projection=ref_src.crs),
+                GeoArray(GeoArray(str(tar_path)), projection=tar_src.crs),
+                **kwargs
+            )
 
-            logger.warning('  Could not co-register the data.')
-            return target
+            try:
+                cr.calculate_spatial_shifts()
+            except:
+                logger.warning('  Could not co-register the data.')
+                return target
 
-        shift_info = cr.correct_shifts()
+            # Apply spatial shifts
+            shift_info = cr.correct_shifts()
 
         left = shift_info['updated geotransform'][0]
         top = shift_info['updated geotransform'][3]
-
+        # Update the transform
         transform = (target.gw.cellx, 0.0, left, 0.0, -target.gw.celly, top)
-
         target.attrs['transform'] = transform
 
-        data = shift_info['arr_shifted'].transpose(2, 0, 1)
-
-        ycoords = np.linspace(top-target.gw.cellyh,
-                              top-target.gw.cellyh-(data.shape[1] * target.gw.celly),
-                              data.shape[1])
-
-        xcoords = np.linspace(left+target.gw.cellxh,
-                              left+target.gw.cellxh+(data.shape[2] * target.gw.cellx),
-                              data.shape[2])
-
-        return xr.DataArray(data=da.from_array(data,
-                                               chunks=(target.gw.band_chunks,
-                                                       target.gw.row_chunks,
-                                                       target.gw.col_chunks)),
-                            dims=('band', 'y', 'x'),
-                            coords={'band': target.band.values.tolist(),
-                                    'y': ycoords,
-                                    'x': xcoords},
-                            attrs=target.attrs)
+        # Check if multi-band
+        if len(shift_info['arr_shifted'].shape) > 2:

```

Comment:
This new logic allows single and multi-band co-reg.

Reply:
super cool